app/controllers/api/v1/simulator_controller.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+class Api::V1::SimulatorController < Api::V1::BaseController
+  include SimulatorHelper
+  include ActionView::Helpers::SanitizeHelper
+
+  before_action :authenticate_user!, only: %i[create update edit]
+  before_action :set_project, only: %i[data edit]
+  before_action :set_user_project, only: %i[update]
+  before_action :check_view_access, only: %i[data]
+  before_action :check_edit_access, only: %i[edit update]
+  # skip_before_action :verify_authenticity_token, only: %i[data create update verilogcv]
+
+  def self.policy_class
+    ProjectPolicy
+  end
+
+  # GET api/v1/simulator/:id/edit
+  def edit
+    render json: Api::V1::UserSerializer.new(current_user)
+  end
+
+  # GET api/v1/simulator/:id/data
+  def data
+    @data = ProjectDatum.find_by(project: @project)
+    if @data
+      render json: @data.data, status: :ok
```

Comment:
```suggestion
    projectDatum = ProjectDatum.find_by(project: @project)
    if projectDatum
      render json: projectDatum.data, status: :ok
```

Let's not confuse with `data`, also I don't think we need instance variable, we can opt normal variable. WDYT?

Reply:
yah sure, actually i put the data check so that when the data is empty we get some meaningful response from the api which can be shown in the error message rather than a empty response. 
for the api route do you want me to change it to something like `get_data` or keep `data`